### ISSUE TITLE
'ExcludeVisualStudioCredential default is true' should be mentioned in the DefaultAzureCredential's documentation.

### ISSUE THREAD

**Issue by michalkrzych** [Azure/azure-sdk-for-net#36352]
Hi all,

When trying to use `DefaultAzureCredential` to [Authenticate .NET apps to Azure services during local development using developer accounts](https://learn.microsoft.com/en-us/dotnet/azure/sdk/authentication-local-development-dev-accounts) it is worth knowing that the `ExcludeVisualStudioCredential`'s default is `true`, so when one has to explicitly set the flag to false to be able to auth with the AZ account logged into the VS (Azure Service Authentication).

```csharp
var defaultAzureCredentialTokenProvider = new DefaultAzureCredential(new DefaultAzureCredentialOptions() {
    ExcludeVisualStudioCredential = false,
});
```

It took me a couple of hours to solve work this out so I thought it was worth mentioning and possibly adding to the docs.

PS. [Related issue](https://github.com/dotnet/docs/issues/35396) to enhance the [linked article](https://learn.microsoft.com/en-us/dotnet/azure/sdk/authentication-local-development-dev-accounts).

Thanks & Regards,
Michal

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 5cebdbb1-4d86-4da0-94b9-de833e8c63ef
* Version Independent ID: d3b9f3f0-4b01-fd1a-1308-6517a0856155
* Content: [DefaultAzureCredentialOptions Class (Azure.Identity) - Azure for .NET Developers](https://learn.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredentialoptions?view=azure-dotnet)
* Content Source: [xml/Azure.Identity/DefaultAzureCredentialOptions.xml](https://github.com/Azure/azure-docs-sdk-dotnet/blob/master/xml/Azure.Identity/DefaultAzureCredentialOptions.xml)
* Service: **azure**
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.

**Comment by christothes** [comment]
Hi @michalkrzych - The documentation for `DefaultAzureCredential` does document the order of precedence for the credentials here: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/identity/Azure.Identity#defaultazurecredential . Do you have a suggestion about how this behavior could be documented more clearly?

**Comment by christothes** [comment]
> Hi @michalkrzych - The documentation for `DefaultAzureCredential` does document the order of precedence for the credentials here: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/identity/Azure.Identity#defaultazurecredential . Do you have a suggestion about how this behavior could be documented more clearly?

I re-read your question a few more times, and I realize that I misinterpreted it. I thought you meant that you were expecting the AZ CLI credential to be utilized even though VS credential was enabled. Could you describe how you determined that VisualStudioCredential is being excluded by default? It definitely shouldn't be excluded unless the `ExcludeVisualStudioCredential` option is explicitly set to `true`. You can also validate this via logging.

Details on how to enable logging can be found [here.](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/samples/Diagnostics.md)

**Comment by ORzazade** [comment]
Hi @michalkrzych 
Thank you for the issue, I have spent 1 day to find out that. It has to be mentioned in the documentation. 

Did the solution work for you? 

here is my example:
`

    shareServiceClient = new ShareServiceClient(new Uri("URL"),
        new DefaultAzureCredential(new DefaultAzureCredentialOptions
        {
            ExcludeVisualStudioCredential = false,
        }));


    var shareClient = shareServiceClient.GetShareClient("fileshare");
    var directoryClient = shareClient.GetRootDirectoryClient();

    var exist = directoryClient.Exists();
`

When I set _false_  and call Exist() then I got **_An HTTP header that's mandatory for this request is not specified._** exception. 

> An HTTP header that's mandatory for this request is not specified.
RequestId:5dcd1d18-b01a-0016-597c-8954d8000000
Time:2023-05-18T11:31:06.6647323Z
Status: 400 (An HTTP header that's mandatory for this request is not specified.)
ErrorCode: MissingRequiredHeader
Additional Information:
HeaderName: x-ms-file-request-intent
Content:
﻿<?xml version="1.0" encoding="utf-8"?><Error><Code>MissingRequiredHeader</Code><Message>An HTTP header that's mandatory for this request is not specified.
RequestId:5dcd1d18-b01a-0016-597c-8954d8000000
Time:2023-05-18T11:31:06.6647323Z</Message><HeaderName>x-ms-file-request-intent</HeaderName></Error>
Headers:
Server: Windows-Azure-File/1.0 Microsoft-HTTPAPI/2.0
x-ms-request-id: 5dcd1d18-b01a-0016-597c-8954d8000000
x-ms-client-request-id: 81b4873b-6414-48fc-8d66-f9b71518b751
x-ms-version: 2022-11-02
x-ms-error-code: MissingRequiredHeader
Date: Thu, 18 May 2023 11:31:05 GMT
Content-Length: 305
Content-Type: application/xml

**Comment by michalkrzych** [comment]
Hi @christothes. Thanks for looking into this. 
After playing with this for a bit I discovered that the `ExcludeVisualStudioCredential` property wasn't actually causing the issue so my sincere apologies for causing the confusion in the first place. I'm quite new to this and may have gotten things slightly wrong :|
I've now updated the [related issue](https://github.com/dotnet/docs/issues/35396) with the steps required in order to use the `DefaultAzureCredential` with the default order of precedence for the credentials and without having to set any properties explicitly.
